File: test90_out/pollen.time/TimerManager/TimerManager.c

/*
 * ======== string literals ========
 */

static const char pollen_time_TimerManager__s__fileName[] = "TimerManager.p";
static const char pollen_time_TimerManager__s__1[] = "(total=";
static const char pollen_time_TimerManager__s__0[] = "TimerManager: registering milli timer ";
static const char pollen_time_TimerManager__s__2[] = ")\n";
static const char pollen_time_TimerManager__s__3[] = "TimerManager";

/*
 * ======== function definitions ========
 */

bool pollen_time_TimerManager_addTimer__E( struct pollen_time_Timer*  t ) {
    bool added;
    uint8 i;
    added = true;
    
    for ((i) = 0; (i) < (pollen_time_TimerManager_numMsTimers__V); ++(i)) {
        if ((pollen_time_TimerManager_msTimers__A[(i)]) == null) {
            (pollen_time_TimerManager_msTimers__A[(i)]) = (t);
            break;
        }
    }
    if ((i) == (pollen_time_TimerManager_numMsTimers__V)) {
        (added) = false;
    }
    else if (pollen_time_TimerManager_isActive__E() == false) {
        pollen_time_TimerManager_start__E();
    }
    return(  (added)  );
}

void pollen_time_TimerManager_stop__E() {
    if (local_mcu_TimerMilli_isRunning__E()) {
        local_mcu_TimerMilli_stop__E();
    }
}

void pollen_time_TimerManager_start__E() {
    if (!local_mcu_TimerMilli_isRunning__E()) {
        local_mcu_TimerMilli_start__E();
    }
}

bool pollen_time_TimerManager_isActive__E() {
    return(  local_mcu_TimerMilli_isRunning__E()  );
}

void pollen_time_TimerManager_targetInit__I() {
}

void pollen_time_TimerManager_tickISR__I() {
    bool oneActive;
    uint8 i;
    pollen_time_Timer_ t;
    oneActive = false;
    for (i = 0; (i) < (pollen_time_TimerManager_numMsTimers__V); ++(i)) {
        if ((pollen_time_TimerManager_msTimers__A[(i)]) != null) {
            t = null;
            (t) = (pollen_time_TimerManager_msTimers__A[(i)]);
            if (pollen_time_Timer_Timer_isActive__E(t)) {
                pollen_time_Timer_Timer_tick__E(t);
                (oneActive) = true;
            }
        }
    }
    if (!(oneActive)) {
        pollen_time_TimerManager_stop__E();
    }
}

File: test90_out/pollen.time/TimerManager/TimerManager.js

/*
 * ======== MODULE pollen.time.TimerManager ========
 */

var $$c = function() {
    const TimerManager = {};
    TimerManager.TimerManager = TimerManager
    TimerManager.$name = 'pollen.time.TimerManager';
    TimerManager.pollen$used = 0;
    TimerManager.Timer = $units['pollen.time.Timer'];
    TimerManager.targetInit = new $$Ref('pollen_time_TimerManager_targetInit__I');
    TimerManager.addTimer = new $$Ref('pollen_time_TimerManager_addTimer__E');
    TimerManager.start = new $$Ref('pollen_time_TimerManager_start__E');
    TimerManager.stop = new $$Ref('pollen_time_TimerManager_stop__E');
    TimerManager.isActive = new $$Ref('pollen_time_TimerManager_isActive__E');
    TimerManager.tickISR = new $$Ref('pollen_time_TimerManager_tickISR__I');
    TimerManager.$$hostInit = function() {
        var $$text = '';
        $units['local.mcu.TimerMilli'].setHandlerOnHost((TimerManager.tickISR));
        (TimerManager.numMsTimers) = 0;
        return $$text;
    }
    TimerManager.registerTimerOnHost = function( t ) {
        var $$text = '';
        (TimerManager.numMsTimers) += 1;
        $$printf("TimerManager: registering milli timer ", "(total=", (TimerManager.numMsTimers), ")\n");
        return $$text;
    }
    TimerManager.pollen__uses__ = function() {
        $$bind($units['pollen.time.Timer'], 'pollen$used', true);
    }
    TimerManager.$$privateInit = function() {
        TimerManager.hwTimer = 0;
        TimerManager.numMsTimers = 1;
    }
    TimerManager.$$hostInitLastPass = function() {
        TimerManager.msTimers = new $$Array((TimerManager.numMsTimers), function($$cn,$$idx){ return 0;}, 'pollen_time_TimerManager_msTimers__A', false); TimerManager.msTimers.$$assign((new $$Array((TimerManager.numMsTimers), function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ null ]));
    }
    return TimerManager;
}
var $$u = $$c();
$units['pollen.time.TimerManager'] = $$u;
$units.push($$u);

